common/nymcoconut/src/scheme/mod.rs

```diff
@@ -0,0 +1,600 @@
+// Copyright 2021 Nym Technologies SA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: implement https://crates.io/crates/signature traits?
+
+use std::convert::TryFrom;
+use std::convert::TryInto;
+
+use bls12_381::{G1Projective, G2Prepared, G2Projective, Scalar};
+use group::Curve;
+
+pub use keygen::{SecretKey, VerificationKey};
+
+use crate::elgamal::Ciphertext;
+use crate::error::{CoconutError, Result};
+use crate::scheme::setup::Parameters;
+use crate::scheme::verification::check_bilinear_pairing;
+use crate::traits::{Base58, Bytable};
+use crate::utils::try_deserialize_g1_projective;
+use crate::{elgamal, Attribute};
+
+pub mod aggregation;
+pub mod issuance;
+pub mod keygen;
+pub mod setup;
+pub mod verification;
+
+pub type SignerIndex = u64;
```

Comment:
If you have any ideas on how to rename it (and all uses of it), I think that'd be the perfect opportunity to do it. I used `index` because it was the first thing that came to my mind when I initially wrote it, but I hate that name so much : (
Just throwing ideas around here: maybe something like `PolynomialCoordinate`? But then `Coordinate` implies (x,y) values while this... thing is just the `x` coordinate used for the Lagrange interpolation...

Reply:
I actually think it is a very decent name :)